### PR TITLE
TakePOS Auto Order improvements

### PR DESCRIPTION
**Pull request by andreubisquerra**

### CODE REVIEW

File: htdocs/takepos/admin/bar.php [hunk 1]
```diff
@@ -155,9 +155,9 @@
         print '<tr class="oddeven value"><td>';
 		print $langs->trans("Table")." ".$row['label'];
 		print '<td>';
-		print "<a target='_blank' href='".$urlwithroot."/takepos/public/auto_order.php?key=".dol_encode($row['rowid'])."'>".$urlwithroot."/takepos/public/auto_order.php?key=".dol_encode($row['rowid'])."</a>";
+		print "<a target='_blank' href='".$urlwithroot."/takepos/auto_order.php?key=".dol_encode($row['rowid'])."'>".$urlwithroot."/takepos/auto_order.php?key=".dol_encode($row['rowid'])."</a>";
```
Comment (eldy) [inline htdocs/takepos/admin/bar.php:158]: Why removing this.
For security purpose, every page that can be accessed without a login/pass must be stored into a directory "public".

**Comment by andreubisquerra** [comment]
Sorry @eldy  I should have mentioned it.
If the auto_order.php file is in public folder the paths in phone.php don't work relatively.
File phone.php require require '../main.inc.php' and if the auto_order.php file is in public folder don't load main.inc.php and crash

May be the soulution is use in phone.php, invoice.php and other files this lines:
if (defined('NOLOGIN')) require '../../main.inc.php';
else require '../main.inc.php';

What is your opinion?

**Comment by eldy** [comment]
i don't think it is enough.
The load of environment (include of main or master), must be by definition done at the higher level. So into the page auto_order.php when you use it.
Also, you can set $_SESSION before the include of main (the session does not exists). 
So i suggest to introduce a constant called INCLUDE_PHONEPAGE_FROM_PUBLIC_PAGE set by auto_order. See 830a117

**Comment by andreubisquerra** [comment]
Nice @eldy !
I will send a new PR with the other improvements

**Pull request closed without merging**